sway-core/src/ir_generation/const_eval.rs

```diff
@@ -118,10 +118,15 @@ pub(crate) fn compile_const_decl(
     // Check if it's a processed global constant.
     match (
         env.module
-            .get_global_constant(env.context, &call_path.as_vec_string()),
+            .get_global_variable(env.context, &call_path.as_vec_string()),
         env.module_ns,
     ) {
-        (Some(const_val), _) => Ok(Some(const_val)),
+        (Some(global_var), _) => {
+            if let Some(constant) = global_var.get_initializer(env.context) {
+                return Ok(Some(Value::new_constant(env.context, *constant)));
+            }
+            Ok(None)
```

Comment:
Can this actually happen? Shouldn't it actually be an ICE? `GlobalVar`s are modeled as variables, means they do not necessarily need to have an initializer, that is fine. My question is the exact semantics of getting an uninitialized `GlobalVar` in the context of `const_eval`. We are compiling a **const** decl and we are getting a `global_var` from the `env.module`. It means it exists, but it doesn't have an initializer.

Doesn't this points to a bug somewhere before in the compilation chain and should be an ICE? Or am I missing some details?